authentik/tenants/models.py

```diff
@@ -85,6 +85,10 @@ class Tenant(TenantMixin, SerializerModel):
     impersonation = models.BooleanField(
         help_text=_("Globally enable/disable impersonation."), default=True
     )
+    impersonation_require_reason = models.BooleanField(
+        help_text=_("Require administrators to provide a reason for impersonating a user."),
+        default=True,
```

Comment:
if we want to release this in 2024.10.x it should probably default to False to be backwards compatible, otherwise this is fine

Reply:
I think it's a good thing to have this default to True as that's what I would expect to be the recommended approach. 

This doesn't have to make it to 2024.10 since it's a new feature